src/internal/internal_gerbt.cc

```diff
@@ -0,0 +1,470 @@
+                }
+                if (ii < mt && jj < nt) {
+                    A22.tileIrecv( ii, jj, A22.tileRank(ii, jj),
+                                  Layout::ColMajor, tag+3, &r );
+                    if (r != MPI_REQUEST_NULL) {
+                        requests.push_back(r);
+                    }
+                }
+            }
+            else {
+                const int64_t compute_rank = A11.tileRank(ii, jj);
+                // Don't need to keep the requests since we don't touch the tile
+                // until recieving the finished data
+                if (jj < nt && A12.tileIsLocal(ii, jj)) {
+                    A12.tileIsend( ii, jj, compute_rank, tag+1, &r );
+                    MPI_Request_free(&r);
+                }
+                if (ii < mt && A21.tileIsLocal(ii, jj)) {
+                    A21.tileIsend( ii, jj, compute_rank, tag+2, &r );
+                    MPI_Request_free(&r);
+                }
+                if (ii < mt && jj < nt && A22.tileIsLocal(ii, jj)) {
+                    A22.tileIsend( ii, jj, compute_rank, tag+3, &r );
+                    MPI_Request_free(&r);
+                }
+            }
+        }
+    }
+    slate_mpi_call(MPI_Waitall(requests.size(), requests.data(),
+                               MPI_STATUSES_IGNORE));
+    requests.clear();
+
+    for (int64_t ii = 0; ii < mt_full; ++ii) {
+        for (int64_t jj = 0; jj < nt_full; ++jj) {
+            if (A11.tileIsLocal(ii, jj)) {
+
+                #pragma omp task shared(A11, A12, A21, A22, U1, U2, V1, V2) \
+                                 firstprivate(ii, jj) priority(1) \
+                                 depend(inout:task[ii*nt_full + jj])
+                {
+                    scalar_t dummy;
+
+                    A11.tileGetForWriting(ii, jj, LayoutConvert::None);
+                    U1.tileGetForReading(ii, 0, LayoutConvert::None);
+                    V1.tileGetForReading(jj, 0, LayoutConvert::None);
+                    Tile<scalar_t> a11 = A11(ii, jj);
+                    Tile<scalar_t> u1 = U1(ii, 0);
+                    Tile<scalar_t> v1 = V1(jj, 0);
+
+                    Tile<scalar_t> a12;
+                    Tile<scalar_t> a21;
+                    Tile<scalar_t> a22;
+                    Tile<scalar_t> u2;
+                    Tile<scalar_t> v2;
+
+                    if (ii < mt) {
+                        A21.tileGetForWriting(ii, jj, LayoutConvert::None);
+                        U2.tileGetForWriting(ii, 0, LayoutConvert::None);
+                        a21 = A21(ii, jj);
+                        u2 = U2(ii, 0);
+                    }
+                    else {
+                        a21 = Tile<scalar_t>(0, a11.nb(), &dummy, 0, HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
+                        u2  = Tile<scalar_t>(0, u1.nb(), &dummy, 0, HostNum,
+                                             TileKind::UserOwned, Layout::ColMajor);
```

Comment:
Why are they declared as `UserOwned`?

Reply:
If I remember right, to prevent SLATE from trying to de-allocate the memory when the tile objects are deleted.  But I'll check whether slate-owned would behave correctly.